packages/coin-avaxc/src/index.ts

```diff
@@ -0,0 +1,538 @@
+  ): Promise<string> => {
+
+    const publicKey = await this.getPublicKey(transport, appPrivateKey, appId, addressIndex);
+
+    const uncompressedKey = ec.keyFromPublic(publicKey, "hex").getPublic(false, 'hex');
+
+    const keyBuffer = Buffer.from(uncompressedKey.substr(2), 'hex');
+
+    const keyHash = createKeccakHash('keccak256').update(keyBuffer).digest('hex');
+
+    const address = "0x".concat(keyHash.substr(-40));
+
+		return address;
+  };
+
+
+  signTransaction = async (
+    transport: Transport,
+    appPrivateKey: string,
+    appId: string,
+    addressIndex: number,
+    transaction: Transaction
+  ): Promise<string> => {
+
+    await apdu.tx.sendScript(transport, params.Transfer.script + params.Transfer.signature);
+
+    const path = await utils.getFullPath({
+      pathType: config.PathType.BIP32,
+      pathString: `44'/${params.PathString}'/0'/0/0`,
+    });
+
+    const argument = '15' + path +
+      handleHex(transaction.to) +
+      handleHex(transaction.value).padStart(20, "0") +
+      handleHex(transaction.gasPrice).padStart(20, "0") +
+      handleHex(transaction.gasLimit).padStart(20, "0") +
+      handleHex(transaction.nonce).padStart(16, "0") +
+      handleHex(params.ChainId.toString(16)).padStart(4, "0") +
+      handleHex(transaction.data);
+
+    const encryptedSig = await apdu.tx.executeScript(
+      transport,
+      appId,
+      appPrivateKey,
+      argument
+    );
+
+    await apdu.tx.finishPrepare(transport);
+
+    await apdu.tx.getTxDetail(transport);
+
+    const decryptingKey = await apdu.tx.getSignatureKey(transport);
+
+    await apdu.tx.clearTransaction(transport);
+
+    await apdu.mcu.control.powerOff(transport);
+
+    const sig = tx.util.decryptSignatureFromSE(encryptedSig!, decryptingKey);
+
+    const { signedTx } = await apdu.tx.getSignedHex(transport);
+
+
+    const rawTx = getRawTx(transaction);
+
+    const rawData = Buffer.from(rlp.encode(rawTx));
+
+    if (rawData.toString('hex') !== signedTx) {
+      throw new Error('unexpected transaction format!');
+    }
+
+    const hash = createKeccakHash('keccak256').update(rawData).digest('hex');
+
+    const data = Buffer.from(handleHex(hash), 'hex');
+
+    const publicKey = await this.getPublicKey(
+      transport, appPrivateKey, appId, addressIndex
+    );
+
+    const keyPair = ec.keyFromPublic(publicKey, 'hex');
+
+    const recoveryParam = ec.getKeyRecoveryParam(data, sig, keyPair.pub);
+
+    const v = recoveryParam + 27;
+    const { r, s } = sig as { r: string; s: string; };
+
+    let vValue = v + params.ChainId * 2 + 8;
+
+    vValue = vValue.toString(16).padStart(6,'0');
+
+    const signedTransaction = rawTx.slice(0, 6);
+
+    signedTransaction.push(
+      Buffer.from(vValue, 'hex'),
+      Buffer.from(r, 'hex'),
+      Buffer.from(s, 'hex')
+    );
+    const serializedTx = Buffer.from(rlp.encode(signedTransaction));
+
+    return `0x${serializedTx.toString('hex')}`;
+  };
+
+
+  signERC20Transaction = async (
```

Comment:
Should automatically apply official signature when data is erc20 transaction and `to` address is a contract address.
FYR
https://github.com/CoolBitX-Technology/coolwallet-sdk/blob/911ca48ed646d9c63b7cb0cc96ea1ca1af39a4dc/packages/coin-matic/src/index.ts#L132-L148